x/marketmap/keeper/keeper.go

```diff
@@ -149,6 +148,19 @@ func (k *Keeper) UpdateMarket(ctx context.Context, market types.Market) error {
 	return k.setMarket(ctx, market)
 }
 
+// DeleteDisabledMarket removes a Market if it is disabled.
+func (k *Keeper) DeleteDisabledMarket(ctx context.Context, tickerStr string) error {
+	// Check if Ticker exists
+	market, err := k.markets.Get(ctx, types.TickerString(tickerStr))
+	if err != nil {
+		return err
+	}
+	if market.Ticker.Enabled {
```

Comment:
There is a bit of asymmetry here since we typically check for Enabled markets for updating in the ante handlers, but here are making it part of policy for all integrations.

Perhaps we should call this method "DeleteMarket", and write something like,

```
for _, handler := k.deleteMarketHandlers {
  if err := handler(ctx, market); err != nil {
    return err
  }
}
return k.markets.Remove(ctx, types.TickerString(tickerStr))
```

We can define a DefaultDeleteMarketHandlers() somewhere and use that to initialize the keeper.

Not sure if handler, verifier, validator is better.

This also ensures if we want to add more complex logic later for a specific integration, we can do so without changing connect.

Reply:
Added the notion of `ValidationHooks` which can be configured for the keeper.

Right now just passing them via options at instantiation which keeps the `NewKeeper` function signature the same